projectile.el

```diff
@@ -1817,36 +1821,38 @@ project-root for every file."
   (let ((prompt (projectile-prepend-project-name prompt))
         res)
     (setq res
-          (cond
-           ((eq projectile-completion-system 'ido)
-            (ido-completing-read prompt choices nil nil initial-input))
-           ((eq projectile-completion-system 'default)
-            (completing-read prompt choices nil nil initial-input))
-           ((eq projectile-completion-system 'helm)
-            (if (and (fboundp 'helm)
-                     (fboundp 'helm-make-source))
-                (helm :sources
-                      (helm-make-source "Projectile" 'helm-source-sync
-                                        :candidates choices
-                                        :action (if action
-                                                    (prog1 action
-                                                      (setq action nil))
-                                                  #'identity))
-                      :prompt prompt
-                      :input initial-input
-                      :buffer "*helm-projectile*")
-              (user-error "Please install helm from \
-https://github.com/emacs-helm/helm")))
-           ((eq projectile-completion-system 'ivy)
-            (if (fboundp 'ivy-read)
-                (ivy-read prompt choices
-                          :initial-input initial-input
-                          :action (prog1 action
-                                    (setq action nil))
-                          :caller 'projectile-completing-read)
-              (user-error "Please install ivy from \
-https://github.com/abo-abo/swiper")))
-           (t (funcall projectile-completion-system prompt choices))))
+          (pcase (if (eq projectile-completion-system 'auto)
+                     (cond
+                      ((and (fboundp 'ido-mode) ido-mode) 'ido)
```

Comment:
That's not a good way to depect it, as the function would always be bound and you're never going to match anything else. I'm not sure if there's a simple way to auto-detect reliably, if someone has istalled several of those packages.

Reply:
Well, it also checks the variable, so this will only trigger when the mode is enabled.

Reply:
@bbatsov I changed fboundp to boundp for more clarity.

Reply:
> Well, it also checks the variable, so this will only trigger when the mode is enabled.

Yeah, this should work, but I think it will also result in compilation warnings as we're now referring to a potentially unbound variable.

Reply:
Forget this. I missed the `defvar`s you've added.